actors/init/src/lib.rs

```diff
@@ -107,6 +111,58 @@ impl Actor {
         Ok(ExecReturn { id_address: Address::new_id(id_address), robust_address })
     }
 
+    /// Exec init actor
+    pub fn exec4<BS, RT>(rt: &mut RT, params: Exec4Params) -> Result<Exec4Return, ActorError>
+    where
+        BS: Blockstore,
+        RT: Runtime<BS>,
+    {
+        if cfg!(feature = "m2-native") {
+            rt.validate_immediate_caller_accept_any()?;
+        } else {
+            rt.validate_immediate_caller_is(iter::once(&EAM_ACTOR_ADDR))?;
+        }
+
+        // Compute the f4 address.
+        let caller_id = rt.message().caller().id().unwrap();
+        let delegated_address =
+            Address::new_delegated(caller_id, &params.subaddress).map_err(|e| {
+                ActorError::illegal_argument(format!("invalid delegated address: {}", e))
+            })?;
+
+        log::trace!("delegated address: {:?}", &delegated_address);
+
+        // Compute a re-org-stable address.
+        // This address exists for use by messages coming from outside the system, in order to
+        // stably address the newly created actor even if a chain re-org causes it to end up with
+        // a different ID.
+        let robust_address = rt.new_actor_address()?;
+
+        log::trace!("robust address: {:?}", &robust_address);
+
+        // Allocate an ID for this actor.
+        // Store mapping of actor addresses to the actor ID.
+        let id_address: ActorID = rt.transaction(|s: &mut State, rt| {
+            s.map_address_to_f4(rt.store(), &robust_address, &delegated_address).map_err(|e| {
+                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to allocate ID address")
```

Comment:
Could you please use the new `context_code` rather than downcast calls for new code? It's on master.